### ISSUE TITLE
Filtering for Global Policy Use

### ISSUE THREAD

**Issue by thewrabbit** [MicrosoftDocs/office-docs-powershell#9371]
Again we're impacted by the migration to 4.x.x Teams PowerShell

We have been unable to successfully filter with get-csonlineuser for users who have a no policy applied i.e. are using the Global policy.  This displays in the full output get-csonlineuser as blank entry against this policy.

Previously there were a number of different methods to return this using PowerShell 2.x.x.

However we've been unable to successfully identify a method using 4.x.x

eg
get-csonlineuser -Filter "TeamsCallingPolicy -ne 'Host:AllowCalling'" returns users with other named policies, not with the Global policy

-like '*' will return all named policies, not the global
-notlike '*' returns no values

We've tried variants of $null, '' with -eq with and without the Authority elements.
We've tried -eq 'Host:Global' and, just in case 'Tenant:Global'

Can the documentation be updated to show how a global policy can be successfully filtered, and if this capability does not exist it's a significant deficit in the PowerShell commands.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7008ec10-a056-d8bb-bb4e-b4690a84adfd
* Version Independent ID: 2dcb78d9-7f36-208e-0989-184905d73011
* Content: [Get-CsOnlineUser (SkypeForBusiness)](https://docs.microsoft.com/en-us/powershell/module/skype/get-csonlineuser?view=skype-ps)
* Content Source: [skype/skype-ps/skype/Get-CsOnlineUser.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/main/skype/skype-ps/skype/Get-CsOnlineUser.md)
* Service: **skypeforbusiness-powershell**
* GitHub Login: @tomkau
* Microsoft Alias: **tomkau**

**Comment by isrumnon** [comment]
Hi @thewrabbit we're looking into this issue and will get back to you soon

**Comment by isrumnon** [comment]
Hi @thewrabbit just to clarify what you mean by no policy applied/global policy - you mean when the input value is null correct?
e.g., Get-CsOnlineUser -Filter "TeamsMeetingPolicy -eq ''"

**Comment by yogkumgit** [comment]
@isrumnon

**Comment by isrumnon** [comment]
HI @thewrabbit can you please retry now?

**Comment by thewrabbit** [comment]
Hi @isrumnon 

Just upgraded to 4.3.0

Tried

Get-CSonlineUser -Filter "TeamsMeetingPolicy -ne 'Staff'"
Only named policies returned (this is actually our preferred command (simplified)

Get-CSOnlineUser -Filter "TeamsMeetingPolicy -eq '""
No results returned.

Get-CSOnlineUser -Filter "TeamsMeetingPolicy -eq 'Global'"
No results returned

So it still seems not to be working - is there a syntax I should be using?

**Comment by isrumnon** [comment]
@thewrabbit apologies there was a small fix that we had to make while using the -eq operator. You can expect this to be enabled by the same time next week

**Comment by thewrabbit** [comment]
Thanks @isrumnon 

Will the -ne operator work too.

We're a little concerned by the timeline here as we need to rework our user creation scripts before the retiral.

**Comment by isrumnon** [comment]
on the -ne operator: this should be working now, did you try this recently?

**Comment by thewrabbit** [comment]
@isrumnon yes tried it after you sent the update today.  Upgraded PS to 4.3.0 and tried


Get-CSonlineUser -Filter "TeamsMeetingPolicy -ne 'Staff'"

This returned users with other named policies, not including users with the global policy.  We're using the policy filter with other filters so if using -ne as an operator we need it to return named and global users.

**Comment by thewrabbit** [comment]
> on the -ne operator: this should be working by end of day, did you try this recently?

@isrumnon  - retested today with -ne operator and still no Global policy users returned.  Only named policies.

**Comment by isrumnon** [comment]
Unfortunately our changes has to be rolled back - can you try again by Friday EOD or latest next Tuesday EOD?

**Comment by thewrabbit** [comment]
Thanks for the updates - checked today and the -ne operator is returning blank entries i.e. global policy users as part of the results now.

Great work - will check on the -eq operator at the end of the week.

Thanks for all your help on this.

**Comment by yogkumgit** [comment]
@isrumnon

**Comment by thewrabbit** [comment]
@isrumnon @yogkumgit 

We're now really concerned that we will be left with an unusable PowerShell module for policy application as of June 15th.

We are a very large tenancy.

- 4.x.x is considerably slower to carry out operations that previous versions - this has a significant operational impact
- We have 7 business days left before we have to use 4.x.x
- 4.x.x is still being worked on by MS to address these issues but we need to time to validate and test our scripts 

We've made what changes we can and have some contingency plans in place but none are particularly satisfactory.

The MC announcement re the retiral of earlier versions was updated today and confirms the mid June date.

**Comment by isrumnon** [comment]
Hi @thewrabbit can you please DM me on Teams or mail me so we can go over possible mitigation paths for your tenant? 
My email - isrumnon@microsoft.com
I'll be deleting this post (for security reasons) assuming you have received an email for the same

**Comment by thewrabbit** [comment]
Alternative workaround identified.

**Comment by isrumnon** [comment]
Hi @thewrabbit, reaching out to you again regarding adoption of modern TPM versions. The updated timelines are now as follows:

- Availability of modern TPM versions (with improved performance/functionality): 15th November
- Deadline for adoption of modern TPM versions: 15th December (tentative)

Modern TPM versions = versions >2.6.0

**Comment by thewrabbit** [comment]
@isrumnon Thanks for the update - we'll start testing as soon as we receive notification that there's an updated version available.  My colleague has been in direct contact with you via Teams so we've allocated resources to the work.

**Comment by isrumnon** [comment]
Hi @yogkumgit no documentation changes required at the present moment

**Comment by yogkumgit** [comment]
Thank you. I proceed to close it, as we are limited to documentation fixes and improvements only.
